app/controllers/task_controller.rb

```diff
@@ -17,24 +19,69 @@ def new
   end
 
   def create
-    @task = Task.new(create_params[:task])
-    @task.save
+     # Guard clause against empty tasknames
+    @taskname = params[:task][:taskname]
+     #Checks if string is empty, sets it to null to trigger db-level error
+    if @taskname.length == 0
```

Comment:
I think now that you have more information about model validations - you'd want to validate this in the model rather than in this controller method